editor/plugins/gizmos/collision_shape_3d_gizmo_plugin.cpp

```diff
@@ -355,6 +356,12 @@ void CollisionShape3DGizmoPlugin::redraw(EditorNode3DGizmo *p_gizmo) {
 		}
 	}
 
+	const bool show_collision_shapes_only_when_selected = EDITOR_GET("editors/3d_gizmos/gizmo_settings/show_collision_shapes_only_when_selected");
```

Comment:
It would be more efficient to cache the value of `show_collision_shapes_only_when_selected` and only update when settings change.